lib/rspec/matchers/english_phrases.rb

```diff
@@ -0,0 +1,58 @@
+module RSpec
+  module Matchers
+    # Facilitates converting ruby objects to English phrases.
+    module EnglishPhrases
+      class << self
+        # Converts a symbol into an English expression.
+        #
+        #     humanize(:banana_creme_pie) #=> "banana creme pie"
+        #
+        # The method name is intended to be familiar to
+        # ActiveSupport users.  It is similar, but has
+        # fewer features.
+        #
+        def humanize(sym)
+          sym.to_s.gsub(/_/, ' ')
+        end
+
+        # Converts an object (often a collection of objects)
+        # into an English list.
+        #
+        #     list(['banana', 'kiwi', 'mango'])
+        #     #=> " \"banana\", \"kiwi\", and \"mango\""
+        #
+        def list(obj)
+          return " #{obj.inspect}" if !obj || Struct === obj
+          items = Array(obj).map { |w| list_item(w) }
+          case items.length
+          when 0
+            ""
+          when 1
+            " #{items[0]}"
+          when 2
+            " #{items[0]} and #{items[1]}"
+          else
+            " #{items[0...-1].join(', ')}, and #{items[-1]}"
+          end
+        end
+
+      private
+
+        def is_matcher_with_description?(object)
+          RSpec::Matchers.is_a_matcher?(object) &&
+            object.respond_to?(:description)
+        end
+
+        # Converts the given item to string suitable for use in a
+        # list expression.
+        def list_item(obj)
```

Comment:
I think I'd prefer this be called `item_description`.  The caller is `list `but there's nothing about this logic that is limited to lists.